pandas/core/arrays/datetimelike.py

```diff
@@ -79,19 +85,270 @@ def _get_attributes_dict(self):
         """return an attributes dict for my class"""
         return {k: getattr(self, k, None) for k in self._attributes}
 
+    @property
+    def _scalar_type(self):
```

Comment:
high-level: why are these new attributes/methods needed?

Reply:
This is to ease code-reuse between the various arrays. `DatetimeLikeArray.__setitem__` is a good example.